### DISCUSSION TITLE
Is Product Grid support  "pimcore/pimcore": "^10.5" ?

### DISCUSSION THREAD

**Discussion by rasikaVillvay** [pimcore/pimcore#16164] [Q&A / Support]
Once i implemented data grid in my pimcore application it's getting error saying 

An exception has been thrown during the rendering of a template ("Cannot resolve argument $segmentTrackingHelperService of "App\Controller\ProductController::listingAction()": Cannot autowire service "App\Services\SegmentTrackingHelperService": argument "$visitorInfoStorage" of method "__construct()" has type "Pimcore\Bundle\PersonalizationBundle\Targeting\VisitorInfoStorage" but this class was not found."). File: /var/www/html/templates/areas/product-grid/view.html.twig Line: 25

**Comment by kingjia90** [comment]
>  "Pimcore\Bundle\PersonalizationBundle\Targeting\VisitorInfoStorage" 

this suggests me you are trying to use a bundle for Pimcore 11, which was moved on https://github.com/pimcore/pimcore/issues/12523, and it shouldn't be used with "^10.5"

**Reply by kingjia90** [reply]
So you are using the Pimcore 11 Demo files 
https://github.com/pimcore/demo/blob/9acc78192e2a6d3a7d22bae59de1c309c2ea3923/src/Services/SegmentTrackingHelperService.php#L23
but if you are on 10.5, it should look like 
https://github.com/pimcore/demo/blob/e5fac9a31b20c48af1c3bdadfff04b2af05e5cca/src/Services/SegmentTrackingHelperService.php#L23

**Reply by rasikaVillvay** [reply]
Thank you i'll try

**Reply by rasikaVillvay** [reply]
Hi,
there are some files are missing
EG: CustomerManagementFrameworkBundle\Targeting\SegmentTracker" but no such service exists.

In this repo

https://github.com/pimcore/demo/blob/e5fac9a31b20c48af1c3bdadfff04b2af05e5cca/src/Services/SegmentTrackingHelperService.php#L23 whare can i get those files

**Reply by kingjia90** [reply]
That's https://github.com/pimcore/customer-data-framework  and https://github.com/pimcore/pimcore/blob/10.6/lib/Targeting/VisitorInfoStorage.php  (which would be there if you are using ^10.5) but not sure what you are trying to achieve and how did you installed 🤔 wouldn't easier switching to demo https://github.com/pimcore/demo/releases/tag/v10.3.5 or so?

**Reply by rasikaVillvay** [reply]
@kingjia90  my pimcore already exist version is 10.5 , for that existing project i am trying to enalble the custome template (Landing page ) component. once i trying to replace the component i am getting this issues on prodouct grid component

**Reply by kingjia90** [reply]
Looks like you are replacing with the wrong file of the Demo, https://github.com/pimcore/demo/tree/10.3 is what you are looking for. 
You replaced with pimcore/demo 11.x and that's not compatible with pimcore/pimcore 10.5

But other than that, i am afraid that can't help much, it's going to be out of the scope, to answer the initial question, it is still supported, is just that your project current state has some incompatibilities and conflicts.

**Reply by rasikaVillvay** [reply]
Thank you for you response i will check and thanks again for the support
